src/components/sorting/sorting.jsx

```diff
@@ -0,0 +1,145 @@
+import React, {PureComponent, createRef} from 'react';
+import {connect} from 'react-redux';
+import {ActionCreator} from '../../reducer.js';
+import {SortType} from '../../const.js';
+import {sortingTypes} from '../../types/types.js';
+
+const sortItems = [
+  {
+    type: SortType.POPULAR,
+    text: `Popular`,
+  },
+  {
+    type: SortType.TO_HIGH,
+    text: `Price: low to high`,
+  },
+  {
+    type: SortType.TO_LOW,
+    text: `Price: high to low`,
+  },
+  {
+    type: SortType.TOP_RATED,
+    text: `Top rated first`,
+  },
+];
+
+class Sorting extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: false,
+    };
+
+    this._handleMenuClick = this._handleMenuClick.bind(this);
+
+    this._selectRef = createRef();
+  }
+
+  componentDidUpdate() {
+    const selectElement = this._selectRef.current;
+
+    if (!selectElement) {
+      return;
+    }
+
+    const {activeSorting} = this.props;
+    selectElement.value = activeSorting;
+  }
+
+  _handleMenuClick() {
+    this.setState((prevState) => ({
+      isOpen: !prevState.isOpen,
+    }));
+  }
+
+  _handleSortClick(sortType) {
+    this.setState({
+      isOpen: false,
+    });
+
+    this.props.onSortItemClick(sortType);
+
+    const selectElement = this._selectRef.current;
+
+    if (!selectElement) {
+      return;
+    }
+    const {activeSorting} = this.props;
+    selectElement.value = activeSorting;
```

Comment:
```suggestion
    const {activeSorting, onSortItemClick} = this.props;
    const selectElement = this._selectRef.current;
    
    this.setState({
      isOpen: false,
    });
    onSortItemClick(sortType);

    if (!selectElement) {
      return;
    }
    selectElement.value = activeSorting;
```
а почему бы не добавить selectElement.value в стейт и не изменять его значение через https://reactjs.org/docs/forms.html#controlled-components